core/kernel/tee_ta_manager.c

```diff
@@ -481,6 +534,9 @@ TEE_Result tee_ta_open_session(TEE_ErrorOrigin *err,
 		return res;
 	}
 
+	if (!check_params(s, param))
+		return TEE_ERROR_BAD_PARAMETERS; /* intentional generic error */
```

Comment:
You are right, this comment is not good, and actually useless. I plan to simply remove it. 
Invalid parameters are simply rejected with an "bad parameter" error code.